utils/DeployedContractsHelpers.ts

```diff
@@ -75,52 +121,82 @@ export async function getTreasury(hre: HardhatRuntimeEnvironment): Promise<Signe
 }
 
-export async function getBDStableWethPool(hre: HardhatRuntimeEnvironment, symbol: string) {
+export async function getBDStableWethPool(hre: HardhatRuntimeEnvironment, symbol: string): Promise<BdStablePool> {
   const deployer = await getDeployer(hre);
-  return (await hre.ethers.getContract(bdstablesContractsDetails[symbol].pools.weth.name, deployer)) as BdStablePool;
+  return (await hre.ethers.getContract(bdStablesContractsDetails[symbol].pools.weth.name, deployer)) as BdStablePool;
 }
 
-export async function getBDStableWethStakingRewards(hre: HardhatRuntimeEnvironment, stableSymbol: string) {
-  return await getBDStableCollateralStakingRewards(hre, stableSymbol, "WETH");
+export async function getStakingRewardsWithWeth(hre: HardhatRuntimeEnvironment, symbol: string): Promise<StakingRewards | undefined> {
+  return await getBDStableCollateralStakingRewards(hre, symbol, "WETH");
 }
 
-export async function getBDStableWbtcStakingRewards(hre: HardhatRuntimeEnvironment, stableSymbol: string) {
-  return await getBDStableCollateralStakingRewards(hre, stableSymbol, "WBTC");
+export async function getStakingRewardsWithWbtc(hre: HardhatRuntimeEnvironment, symbol: string): Promise<StakingRewards | undefined> {
+  return await getBDStableCollateralStakingRewards(hre, symbol, "WBTC");
 }
 
-export async function getBDStableBdxStakingRewards(hre: HardhatRuntimeEnvironment, stableSymbol: string) {
-  return await getBDStableCollateralStakingRewards(hre, stableSymbol, "BDX");
+export async function getStakingRewardsWithBdx(hre: HardhatRuntimeEnvironment, symbol: string): Promise<StakingRewards | undefined> {
+  return await getBDStableCollateralStakingRewards(hre, symbol, "BDX");
 }
 
-async function getBDStableCollateralStakingRewards(hre: HardhatRuntimeEnvironment, stableSymbol: string, collateralSymbol: string) {
-  const deployer = await getDeployer(hre);
-
+async function getBDStableCollateralStakingRewards(
+  hre: HardhatRuntimeEnvironment,
+  stableSymbol: string,
+  collateralSymbol: string
+): Promise<StakingRewards | undefined> {
   const collateralAddress = await getContratAddress(hre, collateralSymbol);
   const stableAddress = await getContratAddress(hre, stableSymbol);
   const poolKey = getPoolKey(collateralAddress, stableAddress, collateralSymbol, stableSymbol);
 
-  return (await hre.ethers.getContract(`StakingRewards_${poolKey}`, deployer)) as StakingRewards;
+  const stakingRewardsContract = await getContractOrUndefined(hre, `StakingRewards_${poolKey}`);
+  return stakingRewardsContract && (stakingRewardsContract as StakingRewards);
 }
 
-export async function getAllBDStableStakingRewards(hre: HardhatRuntimeEnvironment) {
+async function getContractOrUndefined(hre: HardhatRuntimeEnvironment, contractName: string): Promise<Contract | undefined> {
+  try {
+    const deployer = await getDeployer(hre);
+    return hre.ethers.getContract(contractName, deployer);
+  } catch {
+    return undefined;
```

Comment:
I'm not sure. Should it be null or undefined? This always confuses me in JS

Reply:
thanks to your comment I found that ethers has a native `getContractOrNull` function :)